queries.sql

```diff
@@ -8,3 +8,15 @@ select name, escape_attempts from animals where weight_kg > 10.5;
 select * from animals where neutered = '1';
 select * from animals where name != 'Gabumon';
 select * from animals where weight_kg >= 10.4 and weight_kg <= 17.3;
+select count(*) from animals;
+select count(*) from animals where escape_attempts = 0;
+select avg(weight_kg) from animals;
+select avg(escape_attempts) from animals where neutered = '0';
+select avg(escape_attempts) from animals where neutered = '1';
+select min(weight_kg) from animals where name like '%mon%';
+select max(weight_kg) from animals where name like '%mon%';
+select min(weight_kg) from animals where name not like '%mon%';
+select max(weight_kg) from animals where name not like '%mon%';
+select avg(escape_attempts) from animals where species = 'Digimon' and date_of_birth > '1990-01-01' and date_of_birth < '2000-01-01';
+select avg(escape_attempts) from animals where species = 'Pokemon' and date_of_birth > '1990-01-01' and date_of_birth < '2000-01-01';
+
```

Comment:
- [ ] Kindly be noticed that you should write those queries even if you execute them on your machine.
- [ ] Kindly be noticed that you should UPDATE the animals' table by setting the species column to unspecified, then roll back the change.
```sql
   BEGIN; -- start transaction

   UPDATE animals
   SET species = 'unspecified'; -- make change

   SELECT species from animals; -- verify that change was made
   ROLLBACK;

   SELECT species from animals; -- verify that change was undone
```

- [ ] Also notice that you should `UPDATE` the animals' table by setting the species column to digimon for all animals that have a name ending in mon.
- [ ] Also `UPDATE` the animals' table by setting the species column to pokemon for all animals that don't have species already set.
```sql
BEGIN;
   UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon'; 
   UPDATE animals SET species = 'pokemon' WHERE species IS NULL; 

   SELECT species from animals; -- verify that change was made
   COMMIT;
   SELECT species from animals; -- verify that change persists after commit
```

- [ ] Also inside a transaction delete all records in the animals' table, then roll back the transaction.
```sql
BEGIN;

   DELETE FROM animals WHERE ;
   SELECT COUNT(*) FROM ANIMALS;

   ROLLBACK;
   SELECT COUNT(*) FROM ANIMALS;
```

- [ ] Also Inside a transaction:
* `DELETE` all animals born after Jan 1st, 2022
* Create a savepoint for the transaction
* `UPDATE` all animals' weight to be their weight multiplied by -1.
* Rollback to the savepoint
* `UPDATE` all animals' weights that are negative to be their weight multiplied by -1.
* Commit transaction
```sql
BEGIN;

    DELETE FROM animals WHERE  date_of_birth >= '2022/01/01' ;
    SAVEPOINT SP1;

    UPDATE animals SET weight_kg = weight_kg * -1; 
    ROLLBACK TO SP1;
    UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0; 

    COMMIT;
```

Reply:
Hey, @ShahierNashaat thank you so much for the effort, I updated the files. It wasn't clear on the requirements that I should put every action there.